src/index.d.ts

```diff
@@ -87,14 +86,16 @@ export type TVVariants<
 
 export type TVCompoundVariants<
   V extends TVVariants<S>,
-  EV extends TVVariants,
   S extends TVSlots,
   B extends ClassValue,
+  EV extends TVVariants<ES>,
+  ES extends TVSlots,
```

Comment:
A bunch of changes in this PR add `ES` (extended slots) since the slot generic is required in a number of types.